packages/yourloops/components/dialogs/pdf-print-options.tsx

```diff
@@ -0,0 +1,290 @@
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React from "react";
+import { useTranslation } from "react-i18next";
+import dayjs, { Dayjs } from "dayjs";
+
+import { useTheme, makeStyles, Theme } from "@material-ui/core/styles";
+
+import DateRangeIcon from "@material-ui/icons/DateRange";
+
+import useMediaQuery from "@material-ui/core/useMediaQuery";
+import Button from "@material-ui/core/Button";
+import Box from "@material-ui/core/Box";
+import Chip from "@material-ui/core/Chip";
+import Dialog from "@material-ui/core/Dialog";
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import InputAdornment from "@material-ui/core/InputAdornment";
+import TextField from "@material-ui/core/TextField";
+import Typography from "@material-ui/core/Typography";
+
+import { CalendarOrientation } from "../date-pickers/models";
+import RangeDatePicker from "../date-pickers/range-date-picker";
+
+export type Presets = "1week" | "2weeks" | "4weeks" | "3months";
+export interface PrintPDFOptions {
+  /** Print start date (ISO day ex: 2022-02-10) */
+  start: string;
+  /** Print end date (ISO day ex: 2022-02-10) */
+  end: string;
+  preset?: Presets;
+}
+interface DialogPDFOptionsProps {
+  open: boolean;
+  /** Oldest available date date (ISO day ex: 2022-02-10) */
+  minDate: string;
+  /** Newest available date date (ISO day ex: 2022-02-10) */
+  maxDate: string;
+  onResult: (options?: PrintPDFOptions) => void;
+}
+const DEFAULT_PRESET: Presets = "4weeks";
+const MAX_SELECTABLE_DAYS = 90;
+
+const printOptionsStyle = makeStyles((theme: Theme) => {
+  return {
+    marginTop: {
+      marginTop: theme.spacing(2),
+    },
+    customRangeText: {
+      marginBottom: theme.spacing(1),
+    },
+    presetButtons: {
+      width: "20%",
+      marginLeft: theme.spacing(1),
+      [theme.breakpoints.down("sm")]: {
+        width: "40%",
+        marginTop: theme.spacing(1),
+      },
+    },
```

Comment:
CSS improvement to have chips using all available space.
```suggestion
presetButtons: {
      width: "calc(25% - 10px)",
      marginTop: theme.spacing(1),
      [theme.breakpoints.down("sm")]: {
        width: "calc(50% - 5px)",
      },
    },
```